### DISCUSSION TITLE
Use for local loading of embedded JPEG thumbnails

### DISCUSSION THREAD

**Discussion by minorbug** [kean/Nuke#778] [Q&A] [answered]
I have a use case for using Nuke's `LazyImage` to show hundreds, or even thousands, of _local_ images. I have this working locally but it has to read the whole image before displaying. These local images are 20-80mp in size, and most have embedded thumbnails. 

Is there a way to modify the loading process to quickly load thumbnails first, then return the image when its fully loaded?

**Comment by kean** [comment, marked as answer]
Hey, I was just looking into these APIs yesterday. Yes, there is a way to fetch the original thumbnails using [ImageRequest.ThumbnailOptions](https://kean-docs.github.io/nuke/documentation/nuke/imagerequest/thumbnailoptions). It would look something like this:

```swift
// Code wasn't tested
func test() async throws {
    let url = URL(fileURLWithPath: "...")

    let thumbnail = try await ImagePipeline.shared.image(for: ImageRequest(url: url, userInfo: [
        .thumbnailKey: {
            var options = ImageRequest.ThumbnailOptions(maxPixelSize: 1200)
            options.createThumbnailFromImageAlways = false
            return options
        }()
    ]))

    let image = try await ImagePipeline.shared.image(for: url)
}
```

I'm not sure why I decided to go with `.createThumbnailFromImageAlways` set to `true` when I added this API. I'm going to revisit it in the next major version.

Please, be aware that there is a known issue with combining thumbnail and non-thumbnail requests https://github.com/kean/Nuke/issues/705 and it's addressed in the upcoming [Nuke 12.6](https://github.com/kean/Nuke/pull/777).

As for the local files, yes, the framework fully supports loading images from local files, but the requests will go through `URLSession`, which is also something to be aware of.

**Reply by idolize** [reply]
Hi @kean! I'm interested in making use of embedded JPG thumbnails with NukeUI - if I do something like this for each image inside a LazyVGrid:

```swift
LazyImage(
    request: ImageRequest(
        url: url,
        userInfo: [
            .thumbnailKey: {
                var options = ImageRequest.ThumbnailOptions(maxPixelSize: 500)
                options.createThumbnailFromImageAlways = false
                options.createThumbnailFromImageIfAbsent = true
                options.createThumbnailWithTransform = true
                return options
            }()
        ]
    )
) { state in
    if let image = state.image {
        image.resizable()
            .frame(
                width: width,
                height: height
            )
    }
}
```

will it automatically use the thumbnail for the images? Is there any way to verify it is working correctly and not falling back to the full sized images?
